ui-tests/tests/widget.test.ts

```diff
@@ -79,6 +79,99 @@ test('test create new connection', async ({ page }) => {
 
 });
 
+
+const labels_in_memory = ['DuckDB', 'SQLite']
+
+
+test('test irrelevant fields in memory db', async ({ page }) => {
+    await displayWidget(page);
+    await page.locator('#createNewConnection').click();
+    for (const label of labels_in_memory) {
+    await page.locator('#selectConnection').selectOption({ label: label });
+    const username = page.locator('#username');
+    expect(await username.count()).toBe(0);
+    const password = page.locator('#password');
+    expect(await password.count()).toBe(0);
+    const host = page.locator('#host');
+    expect(await host.count()).toBe(0);
+    }
+
+});
+
+const field_defaults = [
+  { label: 'DuckDB', connectionName: 'default', database: ':memory:'},
+  { label: 'SQLite', connectionName: 'default', database: ':memory:'},
+  { label: 'PostgreSQL', connectionName: 'default', port: '5432', username: "", password: "", host: "", database: ""},
+  { label: 'MySQL', connectionName: 'default', port: '3306', username: "", password: "", host: "", database: ""},
+  { label: 'MariaDB', connectionName: 'default', port: '3306', username: "", password: "", host: "", database: ""},
+  { label: 'Snowflake', connectionName: 'default', port: '443', username: "", password: "", host: "", database: ""},
+  { label: 'Oracle', connectionName: 'default', port: '1521', username: "", password: "", host: "", database: ""},
+  { label: 'MSSQL',connectionName: 'default',  port: '1433', username: "", password: "", host: "", database: ""},
+  { label: 'Redshift', connectionName: 'default', port: '5439', username: "", password: "", host: "", database: ""},
+
+];
+
+test('test field defaults no existing connection', async ({ page }) => {
+    await displayWidget(page);
+    await page.locator('#createNewConnection').click();
+    for (const data of field_defaults) {
+    const entries = Object.entries(data);
+    await page.locator('#selectConnection').selectOption({ label: data.label });
+    for (const [key, value] of entries) {
+    if (key !== "label") {
+    expect(await page.locator(`#${key}`).evaluate(select => select.value)).toBe(value);
+      }
+    }
```

Comment:
same, move the for loop outside so instead of having a single test, we have multiple tests, also some spacing to make this more readable

Reply:
[fixed](https://github.com/ploomber/jupysql-plugin/pull/73/commits/4c26f45d33efe0d98276164204a0c34cf6a561fa)